<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>APRS2021</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/fontawesome-all.min.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">





background-image: url(figures/NCRR-logos.svg)
background-size: contain
class: title-slide center middle

## High-resolution portability of 245 polygenic scores&lt;br&gt;when derived and applied in the same cohort

&lt;br&gt;

### APRS 2021

&lt;ins&gt;**Florian Privé**&lt;/ins&gt;, Hugues Aschard, Shai Carmi, Lasse Folkersen, Clive Hoggart, Paul F. O’Reilly, Bjarni J. Vilhjálmsson

---

class: center middle inverse

# Introduction

---

### Major strengths of this study using the UK Biobank 

&lt;br&gt;

- A major concern about PGS is their transferability to other ancestries    
(e.g. bad performances in Africans for PGS trained in Europeans)

--

- UKBB is perfect to provide more detailed evidence of this concern

    - UKBB data contains 450K British, 9K South Asians, 2K East Asians and 7K Africans (including Caribbeans), and other (South and East Europe, Middle East, etc.)
    
    - many different phenotypes are available (we use 245)
    
    `\(\Longrightarrow\)` high-resolution ancestry analysis, averaged over many traits
    
--
    
    - individuals all live in the UK and had their genetic and phenotypic information derived **in the same way**
    
    `\(\Longrightarrow\)` circumvent potential bias when using independent studies

---

class: center middle inverse

# Methods

---

### Ancestry grouping

- We show that squared Euclidean distances on PCs are proportional to `\(F_{ST}\)`

- Then we use these distances and country of birth to define 8 (+1) ancestry groups:

&lt;img src="figures/PC-nine-groups.png" width="85%" style="display: block; margin: auto;" /&gt;

---

### Genetic data

&lt;br&gt;

- 1,040,096 HapMap3 variants (but also secondary analyses with high-quality genotyped variants or a larger set of imputed variants)

- Indidivuals used:

    1. Training based on 391,124 individuals of British ancestry&lt;br&gt;and testing in all 9 ancestry groups (20,000 UK + 28,254 non-UK)
    
    2. Training based on 367,063 UK + 24,061 non-UK&lt;br&gt;(391,124 in total `\(\rightarrow\)` same as before)

&lt;br&gt;

&lt;img src="figures/table1-2.JPG" width="98%" style="display: block; margin: auto;" /&gt;

---

### Phenotypic data

- Mapped ICD10 and ICD9 codes to derive 106 phecodes (mostly diseases)

- Used other UKBB fields to also define 111 continuous and 28 binary phenotypes (so, 245 in total)

&lt;img src="figures/pheno-categories.png" width="85%" style="display: block; margin: auto;" /&gt;

---

### Predictive methods

- **Penalized regression** (efficient LASSO, https://doi.org/10.1534/genetics.119.302019) based on *individual-level* training data (genotype dosage matrix + phenotypes)

- **LDpred2-auto** (https://doi.org/10.1093/bioinformatics/btaa1029) on *summary statistics derived from the individual-level* training data (and in-sample LD matrix)

--

&lt;br&gt;

### Predictive accuracy

**Partial correlation** between PGS (x) and phenotype (y), adjusting for covariates (z).

Basically, regress out x and y by z to get x_eps and y_eps, and compute cor(x_eps, y_eps) `\(\rightarrow\)` *signed* measure.

---

class: center middle inverse

# Results

---

### Lasso results in different ancestries

&lt;img src="figures/lasso-ancestry-2.png" width="95%" style="display: block; margin: auto;" /&gt;

.footnote[Percentage in figure title = squared slope (in blue) // same with LDpred2-auto instead of penalized regression or genotyped variants instead of HM3.]

---

### Drop in prediction versus distance

&lt;br&gt;

&lt;img src="figures/ratio-dist-2.png" width="80%" style="display: block; margin: auto;" /&gt;

&lt;!-- &lt;span class="footnote"&gt;Recall: `\(\text{dist}_{PC}^2 \propto F_{ST}\)`&lt;/span&gt; --&gt;

---

### Using more than HapMap3 variants?

#### "Zoomed" GWAS around top HM3 variants for Lipoprotein A 

&lt;img src="figures/zoom_log_lipoA.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Using more than HapMap3 variants for prediction?

&lt;img src="figures/ldpred2-large-1.png" width="85%" style="display: block; margin: auto;" /&gt;

.footnote[Should try using HM3 + top very high significant?]

---

### (Naively) using multiple ancestries in training

&lt;br&gt; 

&lt;img src="figures/lasso_multi_pcor-1.png" width="88%" style="display: block; margin: auto;" /&gt;

---

### Comparing the two methods used

#### Similar, but LASSO is better when prediction is higher, and inversely

&lt;img src="figures/PLR-ldpred2.JPG" width="90%" style="display: block; margin: auto;" /&gt;

---

class: center middle inverse

## Thank you for joining the session

&lt;br/&gt;&lt;br/&gt;

Preprint: [bit.ly/preprint-UKBB-PGS](https://doi.org/10.1101/2021.02.05.21251061)

Presentation: [bit.ly/pres-UKBB-PGS](https://privefl.github.io/UKBB-PGS/APRS2021.html)

&lt;br/&gt;

<i class="fab  fa-twitter "></i> [privefl](https://twitter.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fab  fa-github "></i> [privefl](https://github.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fab  fa-stack-overflow "></i> [F. Privé](https://stackoverflow.com/users/6103040/f-priv%c3%a9)


.footnote[Slides created with R package [**xaringan**](https://github.com/yihui/xaringan)]


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
