%% LyX 1.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english, 12pt]{article}
\usepackage{times}
%\usepackage{algorithm2e}
\usepackage{url}
\usepackage{bbm}
\usepackage[T1]{fontenc}
\usepackage[latin1]{inputenc}
\usepackage{geometry}
\geometry{verbose,letterpaper,tmargin=2cm,bmargin=2cm,lmargin=1.5cm,rmargin=1.5cm}
\usepackage{rotating}
\usepackage{color}
\usepackage{graphicx}
\usepackage{amsmath, amsthm, amssymb}
\usepackage{setspace}
\usepackage{lineno}
\usepackage{hyperref}
\usepackage{bbm}
\usepackage{makecell}
\usepackage{placeins}
\usepackage{subcaption}

%\renewcommand{\arraystretch}{1.8}

%\linenumbers
%\doublespacing
\onehalfspacing
%\usepackage[authoryear]{natbib}
\usepackage{natbib} \bibpunct{(}{)}{;}{author-year}{}{,}

%Pour les rajouts
\usepackage{color}
\definecolor{trustcolor}{rgb}{0,0,1}

\usepackage{dsfont}
\usepackage[warn]{textcomp}
\usepackage{adjustbox}
\usepackage{multirow}
\usepackage{subcaption}
\usepackage{graphicx}
\graphicspath{{../figures/}}
\DeclareMathOperator*{\argmin}{\arg\!\min}

\let\tabbeg\tabular
\let\tabend\endtabular
\renewenvironment{tabular}{\begin{adjustbox}{max width=0.95\textwidth}\tabbeg}{\tabend\end{adjustbox}}

\makeatletter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LyX specific LaTeX commands.
%% Bold symbol macro for standard LaTeX users
%\newcommand{\boldsymbol}[1]{\mbox{\boldmath $#1$}}

%% Because html converters don't know tabularnewline
\providecommand{\tabularnewline}{\\}
\renewcommand*{\arraystretch}{1.2}

\usepackage{babel}
\makeatother


\begin{document}

\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{table}{0}
\renewcommand{\theequation}{S\arabic{equation}}
\setcounter{equation}{0}

\section*{Supplementary Tables and Figures}

\vspace{10em}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h]
	\centering
	\includegraphics[width=\textwidth]{pheno-categories2}
	\caption{Pie chart of the categories of the 245 phenotypes used in this study. A full description of these phenotypes can be downloaded at \url{https://github.com/privefl/UKBB-PGS/blob/main/phenotype-description.xlsx}.}
	\label{fig:pie}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[p]
	\begin{subfigure}{\textwidth}
		\centering
		\includegraphics[width=.8\linewidth]{admixture_8}
		\caption{with $K=8$ components}
	\end{subfigure}\vspace{2em}
	\begin{subfigure}{\textwidth}
		\centering
		\includegraphics[width=.8\linewidth]{admixture_5}
		\caption{with $K=5$ components}
	\end{subfigure}
	\caption{Results of running ADMIXTURE \cite[]{alexander2009fast} on 200 individuals from each of the nine ancestry groups we define in this study.}
	\label{fig:admixture}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h]
	\centering
	\includegraphics[width=0.85\textwidth]{lasso-ancestry-geno}
	\caption{Partial correlation (and 95\% CI) in the UK test set versus in a test set from another ancestry group. Each point represents a phenotype (only 83 of the continuous phenotypes here) and training has been performed with penalized regression on UK individuals (training 1 in table 1) and \textbf{genotyped} variants. The slope (in blue) is computed using Deming regression accounting for standard errors in both x and y, fixing the intercept at 0.
	The square of this slope is provided above each plot, which we report as the relative predictive performance compared to testing in UK.}
	\label{fig:lasso-ancestry-geno}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.85\textwidth]{lasso-ancestry-geno-norel3}
	\caption{Results from identical analyses as in figure \ref{fig:lasso-ancestry-geno} except that we also remove third-degree relatives in the UK Biobank data we use (instead of second-degree and closer before).}
	\label{fig:lasso-ancestry-geno-norel3}
\end{figure}

\begin{figure}[h]
\centering
\includegraphics[width=0.9\textwidth]{ldpred2-ancestry}
\caption{Partial correlation (and 95\% CI) in the UK test set versus in a test set from another ancestry group. Each point represents a phenotype and training has been performed with \textbf{LDpred2-auto} on UK individuals (training 1 in table 1) and HapMap3 variants. The slope (in blue) is computed using Deming regression accounting for standard errors in both x and y, fixing the intercept at 0.
The square of this slope is provided above each plot, which we report as the relative predictive performance compared to testing in UK.}
\label{fig:ldpred2-ancestry}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[htbp]
	\centerline{\includegraphics[width=0.85\textwidth]{ratio-dist-restricted}}
	\caption{Relative predictive performance compared to the UK (ratio of variance explained in one group compared to in the UK group) versus PC distance from the UK. PCA is computed using individuals from test 1 (Table 1), and PC distances are computed using Euclidean distance between geometric medians of the first 32 PC scores of each ancestry group (shown in figure \ref{fig:PCs-restricted}). Relative performance values are the ones reported in figure 2 of the main text. The slope and standard errors are computed internally by function \texttt{geom\_smooth(method = "lm")} of R package ggplot2.}
	\label{fig:ratio-dist2}
\end{figure}

\begin{figure}[htbp]
\centerline{\includegraphics[width=0.95\textwidth]{PC-scores-restricted}}
\caption{PC scores 19 to 40 when PCA is computed using individuals from test 1 (Table 1). PCs 19 to 32 visually capture some population structure, so we use first 32 PCs when computing the PC distances.}
\label{fig:PCs-restricted}	
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{zoom_log_bilirubin}
	\caption{Zoomed Manhattan plot for total \textbf{bilirubin} concentration. The phenotypic variance explained per variant is computed as $r^2 = t^2 / (n + t^2)$, where $t$ is the t-score from GWAS and $n$ is the degrees of freedom (the sample size minus the number of variables in the model, i.e.\ the covariates used in the GWAS, the intercept and the variant). The GWAS includes all variants with an imputation INFO score larger than 0.3 and within a 500Kb radius around the top hit from the GWAS performed in the UK training set and on the HapMap3 variants, represented by a vertical dotted line.}
	\label{fig:zoom-bilirubin}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{top3_log_bilirubin}
	\caption{Effect sizes and variance explained for the top three variants from figure \ref{fig:zoom-bilirubin}.}
	\label{fig:top3-bilirubin}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{top3_log_lipoA}
	\caption{Effect sizes and variance explained for the top three variants from figure 4.}
	\label{fig:top3-lipoA}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{zoom_apoB}
	\caption{Zoomed Manhattan plot for \textbf{apolipoprotein B} concentration. The phenotypic variance explained per variant is computed as $r^2 = t^2 / (n + t^2)$, where $t$ is the t-score from GWAS and $n$ is the degrees of freedom (the sample size minus the number of variables in the model, i.e.\ the covariates used in the GWAS, the intercept and the variant). The GWAS includes all variants with an imputation INFO score larger than 0.3 and within a 500Kb radius around the top hit from the GWAS performed in the UK training set and on the HapMap3 variants, represented by a vertical dotted line.}
	\label{fig:zoom-apoB}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{top3_apoB}
	\caption{Effect sizes and variance explained for the top three variants from figure \ref{fig:zoom-apoB}.}
	\label{fig:top3-apoB}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h]
	\centering
	\includegraphics[width=0.95\textwidth]{multi_with_prscsx}
	\caption{Partial correlation achieved per phenotype (each panel) and per ancestry group (x-axis) when training penalized regressions either with UK individuals only (training 1 in table 1) or when using individuals of multiple ancestries (training 2). We also run PRS-CSx on training 2, grouping the UK, Italy and Poland groups as a common European group, and removing the small Iran group as it does not have a similar ancestry as the four LD references provided for PRS-CSx. PRS-CSx-Eur corresponds to the raw prediction from PRS-CSx corresponding to the European ancestry data, while PRS-CSx-comb-Eur corresponds to the best prediction from the linear combination of the predictions for all four global ancestries on the UK test data. Phecode 174.1: breast cancer; 185: prostate cancer; 250.2: type 2 diabetes; 401: hypertension; 411.4: coronary artery disease.}
	\label{fig:lasso-multi}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{PLR-ldpred2}
	\caption{\textbf{A)} Partial correlations (and 95\% CI) achieved per phenotype (each point) and per ancestry group (each panel) when training either with LASSO or with LDpred2-auto. \textbf{B)} Focusing now on the UK panel from A), each panel represents a range of proportion of causal variants $p$ and points are colored by SNP heritability $h^2$ (estimates from LDpred2-auto). Penalized regression tends to provide better predictive performance than LDpred2 for phenotypes for which partial-$r$ > 0.3, and inversely.}
	\label{fig:plr-ldpred2}
\end{figure}

\begin{figure}[h]
\centering
\includegraphics[width=0.9\textwidth]{sparse-ldpred2}
\caption{Partial correlations achieved per phenotype (each point) and per ancestry group (each panel) when training either with LDpred2-auto or with LDpred2-auto-sparse (sparse option enabled).}
\label{fig:sparse-ldpred2}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.8\textwidth]{sparsity-plr}
	\caption{Proportion of variants with non-zero effects in the penalized regression models for each phenotype (point) versus the proportion of causal variants $p$ estimated from LDpred2-auto, colored by the partial correlation achieved in the UK test set.}
	\label{fig:sparsity-plr}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.8\textwidth]{sparsity-ldpred2}
	\caption{Proportion of variants with non-zero effects in LDpred2-auto-sparse for each phenotype (point) versus the proportion of causal variants $p$ estimated from LDpred2-auto, colored by the SNP heritability $h^2$ estimated from LDpred2-auto.}
	\label{fig:sparsity-ldpred2}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{timings}
	\caption{Computation times for all penalized regression models run using the 1M HapMap3 variants. We recall that we usually run 90 models for each phenotype because we use 9 sets of hyper-parameters and K=10 folds. Computation time is largely quadratic with the number of non-zero effects in the model. It is also dependent on the compute node and the loading of the HPC cluster at the time of running (Figure \ref{fig:timings-ldpred2}).}
	\label{fig:timings-plr}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=0.9\textwidth]{timings-ldpred2}
	\caption{Computation times for fitting LDpred2-auto (with default 1000 burn-in iterations + 500 more + sparse option running 150 more) using the 1M HapMap3 variants. Running times should be the same for all phenotypes, yet we see some variability depending on the node used. Some fitting had to be run again because it exceeded the 12-hour timeout, which happened a few times when and the HPC cluster was particularly crowded.}
	\label{fig:timings-ldpred2}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[htbp]
	\centerline{\includegraphics[width=0.85\textwidth]{compare-MAF}}
	\caption{Comparison between frequencies in the UK Biobank and frequencies in external data.}
	\label{fig:compare-MAF}
\end{figure}

\begin{figure}[htbp]
	\centerline{\includegraphics[width=0.95\textwidth]{diff-MAF}}
	\caption{Differences in MAF between the first 100,000 variants in UK Biobank and external data. These differences (likely errors in UKBB) are hypothetically grouped around errors in the genotyped data that propagated to the imputed data.}
	\label{fig:diff-MAF}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[htbp]
	\centerline{\includegraphics[width=0.9\textwidth]{jews-ref}}
	\caption{First 24 PC scores for the PCA computed in the reference dataset composed of several Jewish and non-Jewish individuals  \cite[]{behar2013no}. Orange triangles represent the Ashkenazi Jews, pink points the Italian and Sephardi Jews, green points the Maghrebian Jews, and blue points the Iranian and Iraqi Jews.}
	\label{fig:jew-ref}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[htbp]
	\centerline{\includegraphics[width=0.9\textwidth]{qc-plot-new-formula}}
	\caption{Comparison of the standard deviations (SD) computed from both genotypes and summary statistics for the 1000 most associated variants with bilirubin concentration. A) uses the previous formula $\text{sd}(\boldsymbol{G_j}) \approx \frac{\text{sd}(\boldsymbol{y})}{\sqrt{n ~ \text{se}(\hat{\gamma}_j)^2}}$ proposed in \cite{prive2020ldpred2} while B) uses the updated formula $\text{sd}(\boldsymbol{G_j}) \approx \frac{\text{sd}(\boldsymbol{y})}{\sqrt{n ~ \text{se}(\hat{\gamma}_j)^2 + \hat{\gamma}_j^2}}$ proposed here, which does one less approximation.
		The slope slightly larger than 1 can be explained by $\text{sd}(\boldsymbol{y}) > \text{sd}(\boldsymbol{\breve{y}})$.}
	\label{fig:new-formula}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\clearpage

\FloatBarrier

% latex table generated in R 3.6.1 by xtable 1.8-4 package
\begin{table}[p]
	\centering
	\begin{tabular}{|c|c|c|c|}
		\hline
		Phenotype & Set of variants & h2 [2.5\%-97.5\%] & p [2.5\%-97.5\%] \\
		\hline
		174.1 & top1M & 0.0889 [0.086-0.092] & 0.0076 [0.00678-0.00841] \\
		174.1 & HM3 & 0.0299 [0.0264-0.0334] & 0.000881 [0.000636-0.00117] \\
		\hline
		185 & top1M & 0.113 [0.109-0.116] & 0.00819 [0.00743-0.00906] \\
		185 & HM3 & 0.0381 [0.0343-0.0423] & 0.000784 [0.000588-0.00105] \\
		\hline
		411.4 & top1M & 0.0641 [0.0624-0.0659] & 0.0152 [0.0138-0.0168] \\
		411.4 & HM3 & 0.0401 [0.0379-0.0422] & 0.00457 [0.00397-0.00526] \\
		\hline
		apoB & top1M & 0.269 [0.265-0.272] & 0.0533 [0.0498-0.0568] \\
		apoB & HM3 & 0.163 [0.16-0.166] & 0.00132 [0.00119-0.00145] \\
		\hline
		height & top1M & 0.482 [0.479-0.486] & 1 [1-1] \\
		height & HM3 & 0.546 [0.541-0.552] & 0.0226 [0.0218-0.0235] \\
		\hline
		log\_bilirubin & top1M & 0.301 [0.267-0.363] & 0.214 [0.195-0.227] \\
		log\_bilirubin & HM3 & 0.361 [0.357-0.365] & 0.000481 [0.000423-0.000545] \\
		\hline
		log\_BMI & top1M & 0.173 [0.171-0.176] & 1 [1-1] \\
		log\_BMI & HM3 & 0.263 [0.26-0.266] & 0.0426 [0.0404-0.0446] \\
		\hline
		log\_lipoA & top1M & 0.696 [0.689-0.702] & 0.0116 [0.011-0.0122] \\
		log\_lipoA & HM3 & 0.34 [0.336-0.345] & 0.000229 [0.000192-0.000268] \\
		\hline
	\end{tabular}
	\caption{Estimates of SNP heritability $h^2$ and proportion of causal variants $p$ from LDpred2-auto, when using either 1,040,096 HapMap3 variants or when prioritizing 1M variants out of 8M+ common variants, for eight phenotypes. Quantiles of all estimates are also reported. Phecode 174.1: breast cancer; 185: prostate cancer; 411.4: coronary artery disease. \label{tab:ldpred2-est-large}}
\end{table}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\FloatBarrier
\clearpage

\bibliographystyle{natbib}
\bibliography{refs}

\end{document}
